skippkg-finder.py

```diff
@@ -0,0 +1,333 @@
+            return True
+        return False
+
+    def get_packagelist(self, project, by_project=True):
+        """
+        Return the list of package's info of a project.
+        If the latest package is from an incident then returns incident
+        package.
+        """
+
+        pkglist = {}
+        packageinfo = {}
+        query = {'expand': 1}
+        root = ET.parse(http_GET(makeurl(self.apiurl, ['source', project],
+                                 query=query))).getroot()
+        for i in root.findall('entry'):
+            pkgname = i.get('name')
+            orig_project = i.get('originproject')
+            is_incidentpkg = False
+            if pkgname.startswith('000') or\
+                    pkgname.startswith('_') or\
+                    pkgname.startswith('patchinfo.') or\
+                    pkgname.startswith('skelcd-') or\
+                    pkgname.startswith('installation-images') or\
+                    pkgname.endswith('-mini'):
+                continue
+            # ugly hack for go1.x incidents as the name would be go1.x.xxx
+            if '.' in pkgname and re.match(r'[0-9]+$', pkgname.split('.')[-1]) and \
+                    orig_project.startswith('SUSE:') and orig_project.endswith(':Update'):
+                is_incidentpkg = True
+                if pkgname.startswith('go1') or pkgname.startswith('bazel0') or \
+                        pkgname.startswith('dotnet') or pkgname.startswith('ruby2'):
+                    if not (pkgname.count('.') > 1):
+                        is_incidentpkg = False
+
+            # if an incident then update the package origin info
+            if is_incidentpkg:
+                orig_name = re.sub(r'\.[0-9]+$', '', pkgname)
+                incident_number = int(pkgname.split('.')[-1])
+                if orig_name in pkglist and pkglist[orig_name]['Project'] == orig_project:
+                    if re.match(r'[0-9]+$', pkglist[orig_name]['Package'].split('.')[-1]):
+                        old_incident_number = int(pkglist[orig_name]['Package'].split('.')[-1])
+                        if incident_number > old_incident_number:
+                            pkglist[orig_name]['Package'] = pkgname
+                    else:
+                        pkglist[orig_name]['Package'] = pkgname
+            else:
+                pkglist[pkgname] = {'Project': orig_project, 'Package': pkgname}
+
+        if by_project:
+            for pkg in pkglist.keys():
+                if pkglist[pkg]['Project'].startswith('SUSE:') and self.is_sle_specific(pkg):
+                    continue
+                if pkglist[pkg]['Project'] not in packageinfo:
+                    packageinfo[pkglist[pkg]['Project']] = []
+                if pkglist[pkg]['Package'] not in packageinfo[pkglist[pkg]['Project']]:
+                    packageinfo[pkglist[pkg]['Project']].append(pkglist[pkg]['Package'])
+            return packageinfo
+
+        return pkglist
```

Comment:
looks like you need similar functionality like https://github.com/openSUSE/openSUSE-release-tools/pull/2623/files#diff-22a4ac30776b144d4e04816a4e9ccfc2f8999d985f675e160c7a21bf88d2a775R52

maybe we can coordinate and submit already that RemotePackage and RemoteProject objects?

Reply:
maybe, but those special handling in my code have to be there, and I can not skip `kernel-livepatch` stuff, then the script can find them and skips them afterward when added to NON_FTP_PACKAGES.group

Reply:
maybe that special name skips can be parametrized, so we all benefit from that code?

Reply:
I didn't look into RemoteProject and RemotePackage closely, the major 'packagelist` in my code stores package's binarylist which would be selected to ftp-tree, eg. `yast2`
if you do `/source/openSUSE:Leap:15.4?view=info` - inherited ones merged, `yast2` is from `SUSE:SLE-15-SP3:Update`, and it wasn't the right place that RPMs from, it is `yast2.19962`, so `SUSE:SLE-15-SP3:Update/yast2.19962` RPMs have to be selected to ftp-tree, the binarylist of `yast2.19962` has stored to that major `packagelist`.

Any other RPM from other `yast2.xxxxx` updates will not be selected in case it does not exist in `SUSE:SLE-15-SP3:Update/yast2.19962`'s binarylist, these `others` will add to `skiplist`.

Reply:
interesting, thanks for explanation